### DISCUSSION TITLE
meta-aws support for dunfell branch

### DISCUSSION THREAD

**Discussion by srikanth-vv** [aws4embeddedlinux/meta-aws#4116] [Q&A]
Hi,

1. In meta-aws dunfell branch, I don't see "scripts/ec2-ami" folder available. Is it not supported for this branch?
2. In create-ec2-ami.sh (master branch), only X86_64, arm64 architectures are supported. Is there nay specific reason for omitting i386?

My custom linux is built in Dunfell branch and for 32bit device and would like to create EC2 instance to run on aws.
Appreciate your response.

**Comment by thomas-roos** [comment]
Hi,
will look into this during the next sprint - should be able to do this the next two weeks.
- I did not try or backport this feature to dunfell, but I will look into that.
- There are no 32bit ec2 instances afaik. But will try to run a 32bit image on x86_64
Cheers, Thomas

**Comment by thomas-roos** [comment]
Hi,
did look into this and I'm having issues with backporting to dunfell branch.
As well as running i686 in a 64bit EC2.
Will provide details later this week, but just to help me better to understand your use case.
You probably need to upgrade to kirkstone some day when dunfell is EOL, right?
Is there any reason your custom Linux can not be build 64bit?

**Reply by srikanth-vv** [reply]
Hi Thomas,
There is a plan to move to kirkstone, but it will take sometime.
Our hardware is still arm32 and we have created emulator for 32 bit X86 platform and trying to host that in aws EC2 instance.
Let me know, if you need any info, happy to share.

Thanks for looking into it.
Srikanth

**Reply by thomas-roos** [reply]
So your emulator is compiled with yocto?
I wonder why not just compile the image to arm64 and run it as a ec2 ami (this is what the ami ec2 feature does usually)
Other options could be just running a ec2 instance with AL2 linux to run your emulator?

**Comment by thomas-roos** [comment]
In general those issus I had when tried to backport:

meta-virtualization:
- cloud-init missing, this has dependencies to

meta-openembedded:
- python3-charset-normalizer
- python3-configobj

poky:
- grub is missing a feature, likely a upgrade? "grub-mkimage: not found" (when building for x86-64)
- systemd-boot issue: "systemd-boot was skipped: incompatible with host aarch64-poky-linux (not in COMPATIBLE_HOST)" (when building for arm64)
- vhdi.wic image commands are missing (this could be added to local.conf)

there too many missing features, recipes on the dunfell branch to just do a quick backport. 

And those issues adding i686 and arm:
# a x86 32bit image on a x86_64 instance
-> seems that a grub 32bit does not work out of the box with efi
-> 32bit efi does not boot on a 64bit EC2?
-> building a 64bit grub efi booting a 32bit image is currently not supported
-> legacy boot (changing efi to legacy on instance and image) does also not work and there is no option to see what's wrong (only possible with efi, nitro)

# a arm 32bit image on a arm64 instance
-> arch-armv8a does not support 32bit by now
-> arch-armv7 does need more configuration and then will probably not boot

# conclusion
-> there are no 32 bit EC2 instance types anymore
-> It's probably possible but requires a lot of work
-> I doubt it's worth it since 32 is dying on arm and x86
